table/table.go

```diff
@@ -227,6 +228,45 @@ func (t Table) doCommit(ctx context.Context, updates []Update, reqs []Requiremen
 	return New(t.identifier, newMeta, newLoc, t.fsF, t.cat), nil
 }
 
+// TimeTravel finds the snapshot that was current as of or right before the given timestamp.
+func (t Table) TimeTravel(timestampMs int64, inclusive bool) *Snapshot {
```

Comment:
Personally not a fan of the name here, could we do something like `SnapshotAsOf`?

Reply:
Thanks @zeroshade for the quick review. I have updated the code